replaced bleu score with rouge score

ashish3586 opened the pull request.

kaustubhdhole commented:
Please resolve conflicts.

sebastianGehrmann commented on evaluation/evaluate_text_generation.py:
```diff
@@ -1,14 +1,18 @@
+def rogue_score(hypotheses, references):
+    total_f1_scores = 0.0
+    for ref, hyp in zip(references, hypotheses):
+        total_f1_scores+=scorer.score(ref, hyp)['rouge1'][2] # [2] for f1 score
```
Why ROUGE-1?

ashish3586 replied:
rouge1 is matching unigrams between hypothesis and reference. We can also use rouge2(bigram) along with rouge1 if needed.

ashish3586 replied:
Hi Sebastian, I have also added "ROUGE-2" and "ROUGE-L" in the evaluation. Please have a look.

sebastianGehrmann replied:
Thanks

sebastianGehrmann commented on evaluation/evaluate_text_generation.py:
```diff
@@ -1,14 +1,18 @@
+    total_f1_scores = 0.0
+    for ref, hyp in zip(references, hypotheses):
+        total_f1_scores+=scorer.score(ref, hyp)['rouge1'][2] # [2] for f1 score
+    return total_f1_scores/len(hypotheses)
```
The previous code supported multi-reference while this one does not.

ashish3586 replied:
yes, earlier library function "score" from "sacrebleu" is taking a list of hypotheses and references as input, "score" function in "rouge" is taking one hypothesis, reference pair as input. Please let me know if somehow we can make it multi-reference?

sebastianGehrmann replied:
Yes, the original ROUGE paper describes how to do this. I implemented this in BIG-bench like this: https://github.com/google/BIG-bench/blob/main/bigbench/api/task_metrics.py#L86

ashish3586 commented:
> I don't think my initial comment was addressed that the current implementation does not support multi-reference which needs to be added. It probably would also make sense to write docstrings and add a test to ensure that multi-reference and single-reference both work.

Apologies for the very very ........... very late response.  I promise this will not happen again. I changed the rouge function for multi-reference. added docstring with function. added test cases for single reference and multi-reference (in the main function). Please have a look.
